pkg/occlient/occlient.go

```diff
@@ -1515,46 +1421,175 @@ func (c *Client) GetOnePodFromSelector(selector string) (*corev1.Pod, error) {
 	return &pods.Items[0], nil
 }
 
-// RsyncPath copies local directory to directory in running Pod.
-func (c *Client) RsyncPath(localPath string, targetPodName string, targetPath string) (string, error) {
-	log.Debugf("Syncing %s to pod %s:%s", localPath, targetPodName, targetPath)
-
-	// TODO: do this without using 'oc' binary
-	args := []string{
-		"rsync",
-		localPath,
-		fmt.Sprintf("%s:%s", targetPodName, targetPath),
-		"--exclude", ".git",
-		"--no-perms",
+// CopyFile copies localFile to the directory in running Pod.
+// if asFile is true, localFile will be file otherwise it is path to a directory
+// copyFiles is list of changed files captured during `odo watch`
+// localFile is ignored if copyFiles is set
+func (c *Client) CopyFile(localFile string, targetPodName string, targetPath string, copyFiles []string) error {
```

Comment:
`localFile` should be named `localPath` as it will indicate that we want to copy the whole directory.

Reply:
no,  first argument should always be a directory. If single file copy is required the last argument should be used